File: TwitchLMChatBot.Models/Provider.cs

using LiteDB;
using System.Text.Json.Serialization;

namespace TwitchLMChatBot.Models
{
    public class Provider : Entity
    {
        public string Name { get; set; }

        public ProviderType Type { get; set; }

        public string BaseUrl { get; set; }

        public string ApiKey { get; set; }

        public bool IsDefault { get; set; }    

        [BsonIgnore]
        public string TypeName => Type switch
        {
            ProviderType.LMStudio => "LM Studio",
            ProviderType.OpenAi => "Open AI",
            _ => "Not defined.",
        };

    }

}

File: TwitchLMChatBot.Client/src/api/types/provider.ts
export type CreateProviderRequest = {
    name: string;
    providerType: 0|1;
    apiKey?: string;
    baseUrl?: string;
    model?: string;
}

export type Provider = {
    "name": string,
    "type": 0|1,
    "baseUrl": string,
    "apiKey": string,
    "isDefault": boolean,
    "typeName": string,
    "id": number
}
